File: data-structure-and-algorithm/basic/sorting/SelectionSort.java
public class SelectionSort {
	public static void swap(int[] array, int x, int y) {
		int tem = array[x];
		array[x] = array[y];
		array[y] = tem;
	}

	public static void sort(int[] array) {
		int n = array.length;
		for (int i = 0; i < array.length - 1; i++) {
			int min = i;
			for (int j = i + 1; j <= array.length - 1; j++) {
				if (array[j] < array[min]) {
					min = j;
				}
			}
			swap(array, i, min);
		}
	}

	public static void main(String[] args) {
		int[] array = new int[]{1, 4, 7, 2, 5, 9, 0, 2, 6};
		sort(array);
		for (int i : array) {
			System.out.println(i + " ");
		}
	}
}

File: data-structure-and-algorithm/basic/sorting/SelectionSort.py
from typing import List


def sort(a: List[int]):
	length = len(a)
	for i in range(length):
		min = i
		for j in range(i + 1, length):
			if a[j] < a[min]:
				min = j
		a[i], a[min] = a[min], a[i]
